src/actions/actions.ts

```diff
@@ -3615,7 +3615,7 @@ abstract class MoveByScreenLine extends BaseMovement {
       let stop = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.end);
 
       // We want to swap the cursor start stop positions based on which direction we are moving, up or down
-      if (start.line < position.line) {
+      if (start.line < position.line && this.movementType === "up") {
```

Comment:
This will be a separate issue, it is too broken to fix with this, V H, V L, Vgjgjgj, and Vgkgkgk all need to work without breaking other screen line stuff